src/main/kotlin/io/kuzzle/sdk/controllers/RealtimeController.kt

```diff
@@ -25,28 +24,23 @@ class RealtimeController(kuzzle: Kuzzle) : BaseController(kuzzle) {
     )
 
     init {
-        kuzzle.protocol.addListener<UnhandledResponseEvent> {
-            val response = Response().apply {
-                fromMap(JsonSerializer.deserialize(it.message) as Map<String?, Any?>)
+        kuzzle.protocol.addListener<RoomMessageEvent> {
+            val response = it.response
+            var sdkInstanceId = ""
+            if (response.Volatile != null) {
+                sdkInstanceId = response.Volatile!!["sdkInstanceId"].toString()
             }
 
-            if (response.error != null && response.error!!.id.equals("security.token.expired")) {
-                kuzzle.protocol.trigger(TokenExpiredEvent())
-            } else {
-                var sdkInstanceId = ""
-                if (response.Volatile != null) {
-                    sdkInstanceId = response.Volatile!!["sdkInstanceId"].toString()
-                }
-
-                val subs: ArrayList<Subscription>? = currentSubscriptions[response.room]
-                if (subs != null) {
-                    val instanceId = sdkInstanceId
-                    subs.forEach {
-                        if (instanceId == kuzzle.instanceId && it.subscribeToSelf || instanceId != kuzzle.instanceId) {
-                            it.handler(response)
-                        }
+            val subs: ArrayList<Subscription>? = currentSubscriptions[response.room]
+            if (subs != null) {
+                val instanceId = sdkInstanceId
```

Comment:
why create new variable ?

Reply:
I didn't write this code, I just rearanged it